src/software/ai/hl/stp/play/penalty_kick_play.cpp

```diff
@@ -14,44 +14,61 @@ std::string PenaltyKickPlay::getName() const
 
 bool PenaltyKickPlay::isApplicable(const World &world) const
 {
-    if (world.gameState().isOurPenalty())
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
+    return (world.gameState().isReadyState() || world.gameState().isSetupState()) &&
+           world.gameState().isOurPenalty();
 }
 
 bool PenaltyKickPlay::invariantHolds(const World &world) const
 {
-    if (world.gameState().isOurPenalty())
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
+    return world.gameState().isOurPenalty() && !world.gameState().isStopped() &&
+           !world.gameState().isHalted();
 }
 
 void PenaltyKickPlay::getNextTactics(TacticCoroutine::push_type &yield)
 {
     auto penalty_shot_tactic = std::make_shared<PenaltyKickTactic>(
         world.ball(), world.field(), world.enemyTeam().goalie(), true);
 
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::BALL);
```

Comment:
:open_mouth:

Reply:
The whitelisting reason is the same as PenaltyKickEnemy